src/message/messageActionSheet.js

```diff
@@ -301,12 +314,24 @@ export const constructMessageActionButtons = ({
   }
   if (
     message.sender_id === ownUser.user_id
+    // if realm allows users to "edit message", show edit option button
+    && realmAllowMessageEditing
     // Our "edit message" UI only works in certain kinds of narrows.
-    && (isStreamOrTopicNarrow(narrow) || isPmNarrow(narrow))
```

Comment:
You should keep this line the same as it was, and just add the `&& updateMessageTimeLimit < realmMessageContentEditLimit)` on the next line. I'm pretty sure that'll do the same thing, and it's a lot clearer.

Reply:
Hmm.. actually it wouldn't be same, here I'm excluding the edit option in only PM narrow but in stream/topic narrow even though time limit exceeds a message sender should have edit message option to **edit topic** atleast and I'm pretty sure we don't need that in PM narrow